## HEALTH DATA ANALYTICS

# 1.1 General Information

Economic indicators are statistics used by analysts to gauge a country's or region economic perfomance and direction.Income level,one of the indicators,is used in classifying different countries into developed (high income),developing (middle income) and underdeveloped (low income). Economic indicator affect a number of areas in the world including population growth,life expectancy,immunisation rates,vaccines coverage and disease burden.

The estimated world's population stands at 8.2 billion.It is projected to increase by  4.5%, 17.8%, and 25.6% by 2030,2050 and 2100 respectively. The developing and underdeveloped regions have like some parts of the asian region like India and Subsaharan African countries like Nigeria are driving close to half of this growth while the developed countries like Japan have recorded a negative growth indicating a decrease in population with an increase in economic growth.

Additionaly,the global life expectancy is 73 %.With developed countries having a 79% life expactancy and underdeveloped a 64% indicating developed country have access to better healthcare infrastructure therefore have an increased life expectancy compared to developing and underdeveloped countries.

Also,approximately 84% of infants globally have had 3 Diphtheria and meassles vaccine.The developed countries recording the highest number of vaccine coverage and immunisation rates of 90% while the developing and underdeveloped have scanty information due to war,conflicts and natural disasters which result in costant movement in some of these countries and less coverage due to the less resources to reach some of these areas and administer the vaccines.




# 1.2  Problem Statement

Economic indicators,particulary income levels significantly dictate the public health outcomes of both developed(high income) and developing countries(low income).Over time,many developing countries have experienced an economic growth which has remodelled their healthcare sector.The rise in gdp has  increased the counties governemtn capacity to invest in health expenditure and therefore cater for healthcare infrastucture which includes hospital beds,physicians and nurses.

In developed countries ,sustainable economic growth has seen to continuous health expenditure in healthcare infrastructure and services thereby contributing to high life expectancy by reducing mortality rate due to a great vaccine coverage  and prenatal and post natal services sanitation.

Contrastingly ,the developing and underdeveloped countries have been on the knife edge where they have reduced gdp and therefore health expenditure resulting in inadeauate health infrastructure and services this has had an adverse effects on the survival rates and therefore the population growth.

Understanding the relationship between economic growth and health is important for generating public health insights which aims at improving health outcomes in both countries.


# 1.2 Objectives

To examine the relationship between income levels and life expectancy across countries.

To evaluate the effect of economic growth (GDP) on healthcare expenditure and healthcare infrastructure development.

To compare immunization and vaccination coverage among developed, developing, and underdeveloped countries.

To assess the association between healthcare investment and health outcomes such as mortality rates and disease burden.

To analyze disparities in access to healthcare services among countries with different income levels.

To determine how economic indicators influence population health and overall quality of life.

To provide policy recommendations for improving healthcare systems and health outcomes through economic development.

## 2.EDA

# 2.1 Imports and Data Loading

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
import plotly.express as px
import statistics as stats
from scipy import stats as st
import geopandas as gpd
from sklearn.preprocessing import MinMaxScaler
import sklearn as skl

In [5]:
Country = pd.read_csv('country_latest.csv')
Health_indicators = pd.read_csv('health_indicators.csv')
indicator_data = pd.read_csv('indicator_metadata.csv')


## 2.2 Data Exploration

2.2.1  Country dataset

In [6]:
Country.info()

<class 'pandas.DataFrame'>
RangeIndex: 7802 entries, 0 to 7801
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   country_code    7802 non-null   str    
 1   country_name    7802 non-null   str    
 2   region          7802 non-null   str    
 3   income_level    7802 non-null   str    
 4   indicator_code  7802 non-null   str    
 5   indicator_name  7802 non-null   str    
 6   value           7802 non-null   float64
 7   year            7802 non-null   int64  
dtypes: float64(1), int64(1), str(6)
memory usage: 1.2 MB


In [7]:
Country.head()

,country_code,country_name,region,income_level,indicator_code,indicator_name,value,year
0,AFG,Afghanistan,"Middle East, North Africa, Afghanistan & Pakistan",Low income,SP.POP.TOTL,"Population, total",42647492.0,2024
1,ALB,Albania,Europe & Central Asia,Upper middle income,SP.POP.TOTL,"Population, total",2377128.0,2024
2,DZA,Algeria,"Middle East, North Africa, Afghanistan & Pakistan",Upper middle income,SP.POP.TOTL,"Population, total",46814308.0,2024
3,ASM,American Samoa,East Asia & Pacific,High income,SP.POP.TOTL,"Population, total",46765.0,2024
4,AND,Andorra,Europe & Central Asia,High income,SP.POP.TOTL,"Population, total",81938.0,2024


In [8]:
Country.describe()

,value,year
count,7.802000e+03,7802.000000
mean,1.038968e+06,2022.641374
std,2.455005e+07,2.916005
min,-5.391262e+00,2000.000000
25%,5.200000e+00,2022.000000
50%,2.870000e+01,2024.000000
75%,8.479167e+01,2024.000000
max,1.450936e+09,2024.000000


In [9]:
Country.dtypes

country_code          str
country_name          str
region                str
income_level          str
indicator_code        str
indicator_name        str
value             float64
year                int64
dtype: object

In [10]:
print(Country.shape)

(7802, 8)


In [11]:
Cat_cols =Country.select_dtypes(include=['object', 'category']).columns
print(Cat_cols)

Index(['country_code', 'country_name', 'region', 'income_level',
       'indicator_code', 'indicator_name'],
      dtype='str')


C:\Users\grace\AppData\Local\Temp\ipykernel_23216\3176187388.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  Cat_cols =Country.select_dtypes(include=['object', 'category']).columns


In [12]:
num_cols = Country.select_dtypes(include=['number']).columns
print(num_cols)

Index(['value', 'year'], dtype='str')


The country dataset has: 7802 rows and 8 columns and 3 data types: strings ,float64 and integer64
                         

2.2.2 Health indicator dataset 

In [13]:
Health_indicators.info()

<class 'pandas.DataFrame'>
RangeIndex: 5275 entries, 0 to 5274
Data columns (total 48 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   country_code                   5275 non-null   str    
 1   country_name                   5275 non-null   str    
 2   region                         5275 non-null   str    
 3   income_level                   5275 non-null   str    
 4   year                           5275 non-null   int64  
 5   population                     5275 non-null   float64
 6   population_growth              5274 non-null   float64
 7   urban_pct                      5275 non-null   float64
 8   life_expectancy                5275 non-null   float64
 9   life_expectancy_male           5275 non-null   float64
 10  life_expectancy_female         5275 non-null   float64
 11  death_rate                     5275 non-null   float64
 12  birth_rate                     5275 non-null   float64
 13 

In [14]:
Health_indicators.head()

,country_code,country_name,region,income_level,year,population,population_growth,urban_pct,life_expectancy,life_expectancy_male,...,stunting_pct,wasting_pct,overweight_children_pct,undernourishment_pct,fertility_rate,adolescent_fertility,prenatal_care_pct,gdp_per_capita,gdp_per_capita_ppp,poverty_rate
0,ABW,Aruba,Latin America & Caribbean,High income,2000,90588.0,1.030817,65.354550,72.939,70.149,...,NaN,NaN,NaN,NaN,1.845,43.729,NaN,20681.023027,30245.706974,NaN
1,ABW,Aruba,Latin America & Caribbean,High income,2001,91439.0,0.935033,65.335114,73.044,70.251,...,NaN,NaN,NaN,NaN,1.813,40.463,NaN,20740.132583,31920.239073,NaN
2,ABW,Aruba,Latin America & Caribbean,High income,2002,92074.0,0.692052,65.282069,73.135,70.364,...,NaN,NaN,NaN,NaN,1.800,38.185,NaN,21307.248251,31888.508737,NaN
3,ABW,Aruba,Latin America & Caribbean,High income,2003,93128.0,1.138229,65.198292,73.236,70.488,...,NaN,NaN,NaN,NaN,1.808,37.807,NaN,21949.485996,32507.084315,NaN
4,ABW,Aruba,Latin America & Caribbean,High income,2004,95138.0,2.135358,65.088653,73.223,70.487,...,NaN,NaN,NaN,NaN,1.819,38.761,NaN,23700.631990,35059.273098,NaN


In [15]:
Health_indicators.dtypes

country_code                         str
country_name                         str
region                               str
income_level                         str
year                               int64
population                       float64
population_growth                float64
urban_pct                        float64
life_expectancy                  float64
life_expectancy_male             float64
life_expectancy_female           float64
death_rate                       float64
birth_rate                       float64
health_expenditure_pct_gdp       float64
health_expenditure_per_capita    float64
physicians_per_1000              float64
hospital_beds_per_1000           float64
nurses_per_1000                  float64
under5_mortality                 float64
neonatal_mortality               float64
maternal_mortality               float64
tb_incidence                     float64
hiv_incidence                    float64
communicable_death_pct           float64
noncommunicable_

In [16]:
Health_indicators.describe()

,year,population,population_growth,urban_pct,life_expectancy,life_expectancy_male,life_expectancy_female,death_rate,birth_rate,health_expenditure_pct_gdp,...,stunting_pct,wasting_pct,overweight_children_pct,undernourishment_pct,fertility_rate,adolescent_fertility,prenatal_care_pct,gdp_per_capita,gdp_per_capita_ppp,poverty_rate
count,5275.000000,5.275000e+03,5274.000000,5275.000000,5275.000000,5275.000000,5275.000000,5275.000000,5275.000000,4514.000000,...,844.000000,843.000000,806.000000,3762.000000,5275.000000,5275.000000,741.000000,5110.000000,4838.000000,1827.000000
mean,2012.000000,3.378305e+07,1.280475,58.716732,70.748244,68.215760,73.394403,8.192702,21.095707,6.222646,...,24.009360,5.896679,6.198015,10.937400,2.797145,51.097317,86.791826,15749.091058,19781.847712,9.492118
std,7.211786,1.323408e+08,1.618914,23.674753,8.701547,8.471691,9.066795,3.412191,10.888048,2.867900,...,15.043393,4.401812,4.499476,11.043652,1.468998,42.893201,16.137098,25074.200336,22391.204701,17.142033
min,2000.000000,9.544000e+03,-11.356645,8.043814,14.665000,12.383000,20.006000,0.841000,4.200000,1.222602,...,0.500000,0.100000,0.000000,2.500000,0.721000,0.505000,16.100000,109.593814,403.983197,0.000000
25%,2006.000000,7.743870e+05,0.297465,39.060949,65.356000,62.722000,67.976000,6.033500,11.746000,4.126287,...,10.800000,2.400000,2.725000,2.500000,1.670000,15.427500,82.500000,1544.567936,3899.860735,0.300000
50%,2012.000000,5.943366e+06,1.175030,60.031069,72.442000,69.217000,75.689000,7.607000,18.172000,5.640688,...,23.800000,4.900000,5.300000,6.300000,2.274000,39.131000,92.800000,5356.365656,11309.514697,1.500000
75%,2018.000000,2.201484e+07,2.210221,77.627285,77.197000,74.500000,80.115000,9.780500,28.806500,7.966407,...,35.325000,8.700000,8.700000,15.600000,3.638500,75.234000,97.600000,20032.321089,28679.803239,9.800000
max,2024.000000,1.450936e+09,21.700343,100.000000,86.497000,84.560000,88.626000,72.483000,53.390000,27.089685,...,64.000000,24.600000,29.600000,72.500000,7.829000,206.357000,100.000000,288001.433369,180939.439450,94.900000


In [17]:
print(Health_indicators.shape)


(5275, 48)


In [18]:
Cat_cols = Health_indicators.select_dtypes(include=['object', 'category']).columns
print(Cat_cols)

Index(['country_code', 'country_name', 'region', 'income_level'], dtype='str')


C:\Users\grace\AppData\Local\Temp\ipykernel_23216\2477902266.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  Cat_cols = Health_indicators.select_dtypes(include=['object', 'category']).columns


In [19]:
num_cols = Health_indicators.select_dtypes(include=['number']).columns
print(num_cols)

Index(['year', 'population', 'population_growth', 'urban_pct',
       'life_expectancy', 'life_expectancy_male', 'life_expectancy_female',
       'death_rate', 'birth_rate', 'health_expenditure_pct_gdp',
       'health_expenditure_per_capita', 'physicians_per_1000',
       'hospital_beds_per_1000', 'nurses_per_1000', 'under5_mortality',
       'neonatal_mortality', 'maternal_mortality', 'tb_incidence',
       'hiv_incidence', 'communicable_death_pct', 'noncommunicable_death_pct',
       'smoking_male', 'smoking_female', 'obesity_pct', 'alcohol_per_capita',
       'diabetes_pct', 'safe_water_pct', 'sanitation_pct', 'handwashing_pct',
       'immunization_measles', 'immunization_dpt', 'immunization_bcg',
       'immunization_pol3', 'immunization_hib3', 'stunting_pct', 'wasting_pct',
       'overweight_children_pct', 'undernourishment_pct', 'fertility_rate',
       'adolescent_fertility', 'prenatal_care_pct', 'gdp_per_capita',
       'gdp_per_capita_ppp', 'poverty_rate'],
      dtype='str

The health indicator dataset has 5275 rows and 26 columns and 3 datatypes namely;string, float 64 and integer64.

2.2.3 Indicator dataset

In [20]:
indicator_data.info()

<class 'pandas.DataFrame'>
RangeIndex: 43 entries, 0 to 42
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   indicator_code  43 non-null     str  
 1   indicator_name  43 non-null     str  
 2   category        43 non-null     str  
 3   unit            43 non-null     str  
 4   description     43 non-null     str  
dtypes: str(5)
memory usage: 6.8 KB


In [21]:
indicator_data.head()

,indicator_code,indicator_name,category,unit,description
0,SP.POP.TOTL,"Population, total",Demographics,count,"Population, total (Demographics)"
1,SP.POP.GROW,Population growth (annual %),Demographics,%,Population growth (annual %) (Demographics)
2,SP.URB.TOTL.IN.ZS,Urban population (% of total),Demographics,%,Urban population (% of total) (Demographics)
3,SP.DYN.LE00.IN,"Life expectancy at birth, total",Demographics,years,"Life expectancy at birth, total (Demographics)"
4,SP.DYN.LE00.MA.IN,"Life expectancy, male",Demographics,years,"Life expectancy, male (Demographics)"


In [22]:
Indicator_data.dtypes

NameError: name 'Indicator_data' is not defined

In [ ]:
Indicator_data.describe()

,Indicator Code,Indicator Name,Category,Unit,Description
count,43,43,43,43,43
unique,43,43,12,9,43
top,SP.POP.TOTL,"Population, total",Demographics,%,"Population, total (Demographics)"
freq,1,1,8,22,1


In [ ]:
print(Health_indicators.columns.tolist())

['country_code', 'country_name', 'region', 'income_level', 'year', 'population', 'population_growth', 'urban_pct', 'life_expectancy', 'life_expectancy_male', 'life_expectancy_female', 'death_rate', 'birth_rate', 'health_expenditure_pct_gdp', 'health_expenditure_per_capita', 'physicians_per_1000', 'hospital_beds_per_1000', 'nurses_per_1000', 'under5_mortality', 'neonatal_mortality', 'maternal_mortality', 'tb_incidence', 'hiv_incidence', 'communicable_death_pct', 'noncommunicable_death_pct', 'smoking_male', 'smoking_female', 'obesity_pct', 'alcohol_per_capita', 'diabetes_pct', 'safe_water_pct', 'sanitation_pct', 'handwashing_pct', 'immunization_measles', 'immunization_dpt', 'immunization_bcg', 'immunization_pol3', 'immunization_hib3', 'stunting_pct', 'wasting_pct', 'overweight_children_pct', 'undernourishment_pct', 'fertility_rate', 'adolescent_fertility', 'prenatal_care_pct', 'gdp_per_capita', 'gdp_per_capita_ppp', 'poverty_rate']


In [ ]:
print(Indicator_data.shape)


(43, 5)


In [ ]:
Cat_cols = Indicator_data.select_dtypes(include=['object', 'category']).columns
print(Cat_cols)

Index([], dtype='str')


In [ ]:
num_cols = Indicator_data.select_dtypes(include=['number']).columns
print(num_cols)

Index([], dtype='str')


The indicator dataset has 43 rows and 5 columns and has only one datatype namely ; string

## 2.3 Data Cleaning

In [ ]:
# Capitalising the column names 

Health_indicators.columns = Health_indicators.columns.str.title()
Country.columns = Country.columns.str.title()
indicator_data.columns = indicator_data.columns.str.title()


In [ ]:
# Removing the _ sign from the column names

Health_indicators.columns = Health_indicators.columns.str.replace('_', ' ')
Country.columns = Country.columns.str.replace('_', ' ')
indicator_data.columns = indicator_data.columns.str.replace('_', ' ')

In [ ]:
# Converting to the correct data types 

Health_indicators = Health_indicators.convert_dtypes()
Country = Country.convert_dtypes()
Indicator_data = indicator_data.convert_dtypes()

In [ ]:

#Removing trailing and leading spaces from column 

Health_indicators.columns = Health_indicators.columns.str.strip()
Country.columns = Country.columns.str.strip()
indicator_data.columns = indicator_data.columns.str.strip()


In [ ]:
# Checking for missing values in the datasets

print(Health_indicators.isnull().sum())
print(Country.isnull().sum())
print(indicator_data.isnull().sum())

Country Code                        0
Country Name                        0
Region                              0
Income Level                        0
Year                                0
Population                          0
Population Growth                   1
Urban Pct                           0
Life Expectancy                     0
Life Expectancy Male                0
Life Expectancy Female              0
Death Rate                          0
Birth Rate                          0
Health Expenditure Pct Gdp        761
Health Expenditure Per Capita     764
Physicians Per 1000              2444
Hospital Beds Per 1000           2377
Nurses Per 1000                  2443
Under5 Mortality                  425
Neonatal Mortality                425
Maternal Mortality                667
Tb Incidence                      218
Hiv Incidence                    1661
Communicable Death Pct           4177
Noncommunicable Death Pct        4177
Smoking Male                     3971
Smoking Fema

In [ ]:
print(Country.isnull().sum())


Country Code      0
Country Name      0
Region            0
Income Level      0
Indicator Code    0
Indicator Name    0
Value             0
Year              0
dtype: int64


In [ ]:
# sum of missing values in indicator_data

print(indicator_data.isnull().sum())

Indicator Code    0
Indicator Name    0
Category          0
Unit              0
Description       0
dtype: int64


In [ ]:
# percentage of missing values in each column

missing_pct = (Health_indicators.isna().sum() / len(Health_indicators) * 100).sort_values(ascending=False)
print(missing_pct)

Obesity Pct                      100.000000
Diabetes Pct                      92.037915
Prenatal Care Pct                 85.952607
Overweight Children Pct           84.720379
Wasting Pct                       84.018957
Stunting Pct                      84.000000
Noncommunicable Death Pct         79.184834
Communicable Death Pct            79.184834
Smoking Female                    75.279621
Smoking Male                      75.279621
Handwashing Pct                   66.236967
Poverty Rate                      65.364929
Physicians Per 1000               46.331754
Nurses Per 1000                   46.312796
Hospital Beds Per 1000            45.061611
Sanitation Pct                    34.236967
Safe Water Pct                    31.924171
Hiv Incidence                     31.488152
Immunization Hib3                 29.232227
Undernourishment Pct              28.682464
Alcohol Per Capita                26.635071
Immunization Bcg                  26.407583
Health Expenditure Per Capita   

In [ ]:
# Dropping columns with more than 50% missing values
threshold = 50

missing_pct = Health_indicators.isnull().mean() * 100

cols_to_drop = missing_pct[missing_pct > threshold].index

Health_indicators.drop(columns=cols_to_drop, inplace=True)

print(f"Dropped {len(cols_to_drop)} columns:")
print(list(cols_to_drop))

Dropped 12 columns:
['Communicable Death Pct', 'Noncommunicable Death Pct', 'Smoking Male', 'Smoking Female', 'Obesity Pct', 'Diabetes Pct', 'Handwashing Pct', 'Stunting Pct', 'Wasting Pct', 'Overweight Children Pct', 'Prenatal Care Pct', 'Poverty Rate']


In [ ]:
# Filling the missing numeric values


num_cols = Health_indicators.select_dtypes(include='number').columns
int_cols = Health_indicators[num_cols].select_dtypes(include=['int64', 'Int64']).columns

Health_indicators[num_cols] = Health_indicators[num_cols].astype("Float64")

Health_indicators[num_cols] = (
    Health_indicators.groupby(["Region", "Income Level"])[num_cols]
    .transform(lambda x: x.fillna(x.median()))
)

Health_indicators[num_cols] = Health_indicators[num_cols].fillna(
    Health_indicators[num_cols].median()
)

for col in int_cols:
    if Health_indicators[col].notna().all() and (Health_indicators[col] % 1 == 0).all():
        Health_indicators[col] = Health_indicators[col].astype("Int64")

Health_indicators[num_cols] = Health_indicators[num_cols].fillna(
    Health_indicators[num_cols].median()
)

In [ ]:
# filling categorical missing values with mode
cat_cols = Health_indicators.select_dtypes(include=['object', 'category']).columns

for col in cat_cols:
    Health_indicators[col] = (
        Health_indicators[col]
        .fillna(Health_indicators.groupby(['Region', 'Income_Level'])[col]
                .transform(lambda x: x.mode().iloc[0] if not x.mode().empty else np.nan))
        .fillna(Health_indicators.groupby('Region')[col]
                .transform(lambda x: x.mode().iloc[0] if not x.mode().empty else np.nan))
        .fillna(Health_indicators.groupby('Income_Level')[col]
                .transform(lambda x: x.mode().iloc[0] if not x.mode().empty else np.nan))
        .fillna(Health_indicators[col].mode().iloc[0])
    )

In [ ]:
Health_indicators.isnull().sum()

Country Code                     0
Country Name                     0
Region                           0
Income Level                     0
Year                             0
Population                       0
Population Growth                0
Urban Pct                        0
Life Expectancy                  0
Life Expectancy Male             0
Life Expectancy Female           0
Death Rate                       0
Birth Rate                       0
Health Expenditure Pct Gdp       0
Health Expenditure Per Capita    0
Physicians Per 1000              0
Hospital Beds Per 1000           0
Nurses Per 1000                  0
Under5 Mortality                 0
Neonatal Mortality               0
Maternal Mortality               0
Tb Incidence                     0
Hiv Incidence                    0
Alcohol Per Capita               0
Safe Water Pct                   0
Sanitation Pct                   0
Immunization Measles             0
Immunization Dpt                 0
Immunization Bcg    

In [ ]:
 # Saving the cleaned data to a new CSV file
Health_indicators.to_csv('cleaned_Health_indicators_data.csv', index=False)

In [ ]:
Country.to_csv('cleaned_country_data.csv', index=False)

In [ ]:
indicator_data.to_csv('cleaned_indicator_data.csv', index=False)

Column names were capitalized and underscore removed from headers to improve readability and for proffesional presentation.

The Columns were converted to correct data types to ensure the data was interpreted and processed accurately.

Trailing spaces were removed from column names  improve data consistent and make joins and merges on datasets easier.

Missing values were checked to prevents inaccurate analysis.Only the columns with more than 50% missing values dropped due to insufficient data which limit draws unrelaible conclusion;12 columns were dropped.The remaining columns with less than 50% were retained.Missing values in the numerical and categorical columns were then filled with the median and mode repectively, within each region and income level to preserve the geographical and economic similarities.Median was used at it is less affected by outliers and the mode used as it represents the most frequently occuring category in the colim and it therefore preserves the categorical variables.

Cleaned datasets were saved to preserve the cleaning steps and prepare it for the analysis and visulisation step.




## 2.4 Data Merging

In [23]:
health = pd.read_csv("cleaned_health_indicators_data.csv")
country = pd.read_csv("cleaned_country_data.csv")
indicator = pd.read_csv("cleaned_indicator_data.csv")

In [24]:
merged_df = health.merge(
country[
["Country Code", "Country Name", "Region", "Income Level"]
].drop_duplicates(),
on="Country Code",
how="left",
suffixes=("", "_country")
)
# Remove duplicated metadata columns
merged_df = merged_df.drop(
columns=[c for c in merged_df.columns if c.endswith("_country")]
)
how="left",
suffixes=("", "_country")

# Remove duplicated metadata columns
merged_df = merged_df.drop(
columns=[c for c in merged_df.columns if c.endswith("_country")]
)
# save merged df as csv
merged_df.to_csv("merged_health_country_data.csv", index=False)

#display the merged dataframe shape and first few rows
print(merged_df.shape)
merged_df.head()

(5275, 36)


,Country Code,Country Name,Region,Income Level,Year,Population,Population Growth,Urban Pct,Life Expectancy,Life Expectancy Male,...,Immunization Measles,Immunization Dpt,Immunization Bcg,Immunization Pol3,Immunization Hib3,Undernourishment Pct,Fertility Rate,Adolescent Fertility,Gdp Per Capita,Gdp Per Capita Ppp
0,ABW,Aruba,Latin America & Caribbean,High income,2000,90588,1.030817,65.354550,72.939,70.149,...,92,92,95,92,92,6.3,1.845,43.729,20681.023027,30245.706974
1,ABW,Aruba,Latin America & Caribbean,High income,2001,91439,0.935033,65.335114,73.044,70.251,...,92,92,95,92,92,6.3,1.813,40.463,20740.132583,31920.239073
2,ABW,Aruba,Latin America & Caribbean,High income,2002,92074,0.692052,65.282069,73.135,70.364,...,92,92,95,92,92,6.3,1.800,38.185,21307.248251,31888.508737
3,ABW,Aruba,Latin America & Caribbean,High income,2003,93128,1.138229,65.198292,73.236,70.488,...,92,92,95,92,92,6.3,1.808,37.807,21949.485996,32507.084315
4,ABW,Aruba,Latin America & Caribbean,High income,2004,95138,2.135358,65.088653,73.223,70.487,...,92,92,95,92,92,6.3,1.819,38.761,23700.631990,35059.273098


The datasets were merged to combine related information in each separate dataset into a comprehensive dataset to enable detailed analysis and exploration of relationships.

The merging was on country code only as both datsets had the same info on country code and not year because the year column in the country dataset had only 2024 while the health indicator had form 2001 to 2024

## 2.5 Data Visualisation

In [32]:
import plotly.graph_objects as go

# Aggregate both metrics by year
overall = (
    merged_df.groupby("Year")["Life Expectancy"]
    .mean()
    .reset_index()
)

overall_gdp = (
    merged_df.groupby("Year")["Gdp Per Capita Ppp"]
    .mean()
    .reset_index()
)

# Create figure
fig = go.Figure()

# Life Expectancy (left y-axis)
fig.add_trace(
    go.Scatter(
        x=overall["Year"],
        y=overall["Life Expectancy"],
        mode="lines+markers",
        name="Life Expectancy"
    )
)

# GDP per Capita (right y-axis)
fig.add_trace(
    go.Scatter(
        x=overall_gdp["Year"],
        y=overall_gdp["Gdp Per Capita Ppp"],
        mode="lines+markers",
        name="GDP per Capita (PPP)",
        yaxis="y2"
    )
)

# Layout with secondary axis
fig.update_layout(
    title="Average Life Expectancy and GDP per Capita Over Time",
    xaxis=dict(title="Year"),
    yaxis=dict(
        title="Life Expectancy (Years)"
    ),
    yaxis2=dict(
        title="GDP per Capita PPP (USD)",
        overlaying="y",
        side="right"
    ),
    legend=dict(
        x=0.01,
        y=0.99
    )
)

fig.show()

In [44]:
import plotly.express as px

overall_health_exp = (
    merged_df.groupby("Year")["Health Expenditure Pct Gdp"]
    .mean()
    .reset_index()
)

fig = px.line(
    overall_health_exp,
    x="Year",
    y="Health Expenditure Pct Gdp",
    markers=True,
    title="Average Health Expenditure Over Time"
)

fig.update_layout(
    xaxis_title="Year",
    yaxis_title="Health Expenditure (% of GDP)"
)

fig.show()

In [50]:
from sklearn.preprocessing import MinMaxScaler
import plotly.express as px

trend = (
    merged_df.groupby("Year")[
        ["Life Expectancy", "Gdp Per Capita Ppp", "Health Expenditure Pct Gdp"]
    ]
    .mean()
    .reset_index()
)

scaler = MinMaxScaler()

trend_norm = trend.copy()
trend_norm.iloc[:, 1:] = scaler.fit_transform(trend.iloc[:, 1:])

fig = px.line(
    trend_norm,
    x="Year",
    y=[
        "Life Expectancy",
        "Gdp Per Capita Ppp",
        "Health Expenditure Pct Gdp"
    ],
    markers=True,
    title="Normalized Trends: GDP, Life Expectancy and Health Expenditure"
)

fig.update_layout(
    yaxis_title="Normalized Value (0-1)"
)

fig.show()

In [45]:
avg_infra = (
    merged_df.groupby("Year")[
        [
            "Physicians Per 1000",
            "Nurses Per 1000",
            "Hospital Beds Per 1000"
        ]
    ]
    .mean()
    .reset_index()
)

fig = px.line(
    avg_infra,
    x="Year",
    y=[
        "Physicians Per 1000",
        "Nurses Per 1000",
        "Hospital Beds Per 1000"
    ],
    markers=True,
    title="Average Health Infrastructure Indicators Over Time"
)

fig.update_layout(
    xaxis_title="Year",
    yaxis_title="Per 1,000 Population"
)

fig.show()

The average life expectancy and gdp per capita increases steadily with each increasing year till 2020 where a sharp drop is seen before the it rises again in gdp and a further drop in life expectancy in 2021 before it rises again.

The steady increases in life expectancy as the gdp increases indicated that an increase in economic growth indicated by gdp corresponds with an increase in life expectancy.This is because the increase in gdp generally means an increase in health expenditure,therefore more money is allocated to the healthcare infrastructure like hospital beds, physician and nurses.Additionally,there's access to clean and food therefore increases life expectancy.

The sharp drop seen in 2020 was as a result of the COVID 19 pandemic which resulted in a worldwide lockdown which halted and slowed down businesses resulting in a drop in gdp thereby less money was allocated for health expenditure crippling the health sector.Additionally,the life expectancy dropped due to healthcare strain ,delayed treatments and overwehelmed hospitals with less hospital equipments to treat patients and persons with underlysing medical conditions.Also due to the isolation,there was a reported number of increase in drug overdose due to emotional stress as a result of layoffs and business closure which also claimed people lives alongside the pandemic.

In 2021,a further decrease in life expectancy was seen this was due to the aftermath of the 2020 pandemic.After the lockdown was halted and business resumed their operations,a surge in the economic growth was seen resulting in an increase in investment in the healthcare sector and people access to food and sanitation resulting in an increase in life expectancy post 2021.

The graph below further highlight the life expectancy vs gdp disparity per income level.

In [ ]:
import plotly.express as px

fig = px.scatter(
    merged_df,
    x="Gdp Per Capita Ppp",
    y="Life Expectancy",
    animation_frame="Year",
    animation_group="Country Code",
    size="Population",
    color="Income Level",
    hover_name="Country Name",
    log_x=True,
    size_max=60,
    title="Health-Wealth Gradient Over Time"
)

fig.update_layout(
    xaxis_title="GDP Per Capita PPP (Log Scale)",
    yaxis_title="Life Expectancy",
    height=700
)

fig.show()

Life expectancy increases with an increase in gdp

DO IMUNIZATION RATES CORRELATE WITH SURVIVAL?

Mortality rate negatively correlates with immunization inndicating that immunisation postitively correlates survival 

In [38]:
import plotly.express as px

trend_data = (
    merged_df.groupby("Year")[
        [
            "Safe Water Pct",
            "Sanitation Pct",
            "Immunization Dpt",
            "Immunization Measles",
            "Under5 Mortality",
            "Neonatal Mortality",
            "Tb Incidence"
        ]
    ]
    .mean()
)

fig = px.imshow(
    trend_data.T,
    aspect="auto",
    color_continuous_scale="RdYlGn",
    title="Health Indicators Over Time"
)

fig.show()

In [40]:


cols = [
    "Immunization Dpt",
    "Immunization Measles",
    "Safe Water Pct",
    "Sanitation Pct",
    "Under5 Mortality",
    "Neonatal Mortality",
    "Tb Incidence"
]

corr = merged_df[cols].corr()
corr = corr.round(2)

fig = px.imshow(
    corr,
    text_auto=True,
    color_continuous_scale="RdBu_r",
    title="Prevention and Health Outcomes Correlation Matrix"
)

fig.update_layout(
    xaxis_title="Health Indicators",
    yaxis_title="Health Indicators",
    height=700
)

fig.show()


HOW DOES HEALTH CAPACITY IMPACT MARTERNAL MORTALITY?

High income countries have more physician ,nurses and hospital beds therefore record low mortality rates and the reverse in lower income countries where they have fewer healthcare resources  thereby record high maternal moratlity rates.

Thereby highlighting  that the healthcare infrastructure is indirectly proportional to martenal mortality.


In [ ]:
import plotly.graph_objects as go

infra = merged_df.groupby("Income Level").agg({
    "Physicians Per 1000":"mean",
    "Nurses Per 1000":"mean",
    "Hospital Beds Per 1000":"mean",
    "Maternal Mortality":"mean"
}).reset_index()

fig = go.Figure()

fig.add_bar(
    x=infra["Income Level"],
    y=infra["Physicians Per 1000"],
    name="Physicians"
)

fig.add_bar(
    x=infra["Income Level"],
    y=infra["Nurses Per 1000"],
    name="Nurses"
)

fig.add_bar(
    x=infra["Income Level"],
    y=infra["Hospital Beds Per 1000"],
    name="Hospital Beds"
)

fig.add_trace(
    go.Scatter(
        x=infra["Income Level"],
        y=infra["Maternal Mortality"],
        yaxis="y2",
        mode="lines+markers",
        name="Maternal Mortality"
    )
)

fig.update_layout(
    title="Healthcare Capacity vs Maternal Mortality",
    barmode="group",
    yaxis=dict(title="Infrastructure"),
    yaxis2=dict(
        title="Maternal Mortality",
        overlaying="y",
        side="right"
    )
)
fig.update_layout(width=1000, height=800)
fig.show()

In [ ]:
import plotly.graph_objects as go

fig = go.Figure()

fig.add_trace(go.Scatterpolar(
    r=infra_region["Physicians Per 1000"],
    theta=infra_region["Region"],
    fill='toself',
    name='Physicians',
    line_color='blue'
))

fig.add_trace(go.Scatterpolar(
    r=infra_region["Nurses Per 1000"],
    theta=infra_region["Region"],
    fill='toself',
    name='Nurses',
    line_color='green'
))

fig.add_trace(go.Scatterpolar(
    r=infra_region["Hospital Beds Per 1000"],
    theta=infra_region["Region"],
    fill='toself',
    name='Hospital Beds',
    line_color='red'
))

fig.update_layout(
    title="Healthcare Infrastructure by Region",
    polar=dict(
        radialaxis=dict(
            visible=True
        )
    ),
    showlegend=True
)

fig.show()

DOES IMMUNISATION LEVEL CORRELATE WITH SURVIVAL AT DIFFERENT INCOME LEVELS?

~There's a negative correlation .Under 5 mortality decreases with an increase in preventive index .This indicates that high immunisation rates leads to reduced under 5 mortality thereby lowering burden.

In [ ]:
vaccines = [
    "Immunization Measles",
    "Immunization Dpt",
    "Immunization Bcg",
    "Immunization Pol3",
    "Immunization Hib3"
]

merged_df["Preventative Index"] = merged_df[vaccines].mean(axis=1)

latest = merged_df[merged_df["Year"] == merged_df["Year"].max()]

fig = px.scatter(
    latest,
    x="Preventative Index",
    y="Under5 Mortality",
    color="Income Level",
    size="Population",
    hover_name="Country Name",
    trendline="ols",
    title="Preventative Index vs Under-5 Mortality"
)

fig.show()

DOES VULNERABILITY CORRELATE WITH LIFE EXPECTANCY?

Life expectancy decreases with an increase in vulnerability index ,indicating a negative correlation between the two measures.

In [ ]:


cols = [
    "Undernourishment Pct",
    "Adolescent Fertility",
    "Death Rate",
]

scaler = MinMaxScaler()
scaled = scaler.fit_transform(merged_df[cols])

merged_df["Vulnerability Index"] = scaled.mean(axis=1)

import plotly.express as px

latest = merged_df[merged_df["Year"] == merged_df["Year"].max()]

fig = px.scatter(
    latest,
    x="Vulnerability Index",
    y="Life Expectancy",
    color="Income Level",
    size="Population",
    hover_name="Country Name",
    trendline="ols",
    title="Vulnerability vs Life Expectancy",
)

fig.show()


In [ ]:
import plotly.express as px

latest = merged_df[merged_df["Year"] == merged_df["Year"].max()]

fig = px.choropleth(
    latest,
    locations="Country Code",
    color="Life Expectancy",
    hover_name="Country Name",
    color_continuous_scale="Viridis",
    title="Global Life Expectancy"
)

fig.show()

DOES RISING GDP EASE COMMUNICABLE DISEASE BURDEN ?

An increase in GDP results in decrease in TB and HIV incidence indiating that the gdp arise ease the disease burden.





In [ ]:
fig= px.scatter(
    latest,
    x="Gdp Per Capita Ppp",
    y="Tb Incidence",
    color="Region",
    trendline="ols",
    log_x=True
)
fig.show()

fig= px.scatter(
    latest,
    x="Gdp Per Capita Ppp",
    y="Hiv Incidence",
    color="Region",
    trendline="ols",
    log_x=True
)
fig.show()

c:\Users\grace\AppData\Local\Python\pythoncore-3.14-64\Lib\site-packages\statsmodels\regression\linear_model.py:1782: RuntimeWarning: invalid value encountered in scalar divide
  return 1 - self.ssr/self.centered_tss


DOES INCOME LEVEL

In [41]:
import plotly.express as px

fig = px.scatter(
    merged_df,
    x="Gdp Per Capita Ppp",
    y="Health Expenditure Pct Gdp",
    color="Region",
    size="Population",
    hover_name="Country Name",
    title="GDP vs Health Expenditure by Income Level"
)

fig.update_layout(
    xaxis_title="GDP Per Capita PPP",
    yaxis_title="Health Expenditure (% of GDP)",
    height=700,
    width=1000
)

fig.update_xaxes(rangemode="tozero")
fig.update_yaxes(rangemode="tozero")

fig.show()